src/bsp/platform/stm32f4_nucleo_wifi/xi_bsp_io_net_stm32f4_nucleo_wifi.c

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2003-2016, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <stdio.h>
+#include <xi_bsp_io_net.h>
+#include "wifi_interface.h"
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef MAX
+#define MAX( a, b ) ( ( ( a ) > ( b ) ) ? ( a ) : ( b ) )
+#endif
+
+#define XI_BSP_IO_NET_BUFFER_SIZE 512
```

Comment:
In order to make it compatible with xively-client-c this should be equal to already defined - XI_IO_BUFFER_SIZE - but please check my comments about using different API functions.